app/models/xml_source_processor.rb

```diff
@@ -345,34 +345,55 @@ def update_links_and_xml(xml_string, preview_mode=false, text_type)
     # first clear out the existing links
     # log the count of articles before and after
     clear_links(text_type) unless preview_mode
-    processed = ""
+
+    candidate_articles = collection.articles.left_joins(:article_versions)
+    page_update_timestamp = 1.hour.ago
+
+    processed = ''
     # process it
     doc = REXML::Document.new xml_string
     doc.elements.each("//link") do |element|
       # default the title to the text if it's not specified
-      if !(title=element.attributes['target_title'])
+      if !(title = element.attributes['target_title'])
         title = element.text
       end
-      #display_text = element.text
-      display_text = ""
+      # display_text = element.text
+      display_text = ''
       element.children.each do |e|
         display_text += e.to_s
       end
       debug("link display_text = #{display_text}")
-      #change the xml version of quotes back to double quotes for article title
+      # change the xml version of quotes back to double quotes for article title
       title = title.gsub('&quot;', '"')
 
+      article = candidate_articles.find_by(title: title)
+
+      if article.nil?
+        article_version = ArticleVersion.where(article_id: candidate_articles.select(:id))
+                                        .where(title: title)
+                                        .where('created_on > ?', page_update_timestamp)
+                                        .order(created_on: :desc)
+                                        .first
+        article = article_version&.article
```

Comment:
I think that we might be able to use the left join above here by doing something like this:
```ruby
candidate_articles.where('article_versions.title': title)
                                 .where(where('article_versions.created_on > ?', page_update_timestamp)
```
etc.  

It's possible that in that case we'd get the correct article, rather than having to pluck it from the article_version, since it would already be in the join.